### PR TITLE
Add support for licensing in the tests

### PR DESCRIPTION
**Pull request by adamtheturtle**
Licensing is soon to land in DC/OS Enterprise master. This change means that the E2E test suite will not break when that happens.

Travis CI is [currently having problems](https://www.traviscistatus.com/incidents/c1s4dlyxd4lf) so I would like a review modulo CI.

This removes the nice styling and comments from the Travis configuration file but that allows us to use tooling to update the configuration file. I moved many of the comments to the contributing documentation.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -102,3 +102,6 @@ node_modules/
 
 # NPM
 package-lock.json
+
+# DC/OS License
+license-key.txt
```
Comment (jimenez) [inline .gitignore]: s/license-key.txt/license.txt

File: .travis.yml [hunk 2]
```diff
@@ -1,62 +1,42 @@
...
 before_install:
-    - "travis_retry pip install --upgrade pip setuptools codecov"
-    - "travis_retry sudo apt-get update"
-      # Ideally we would use `addons.apt.packages` for enchant but this is not
-      # supported by the linter.
-    - "travis_retry sudo apt-get install docker-ce enchant"
-    - "travis_retry npm -g install npm@latest"
-
+- openssl aes-256-cbc -K $encrypted_0ff8d3358b65_key -iv $encrypted_0ff8d3358b65_iv
+  -in license-key.txt.enc -out license-key.txt -d
```
Comment (jimenez) [inline .travis.yml]: s/license-key.txt/license.txt

File: .travis.yml [hunk 3]
```diff
@@ -1,62 +1,42 @@
...
+- cp license-key.txt /tmp/license-key.txt
```
Comment (jimenez) [inline .travis.yml]: s/license-key.txt/license.txt

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -86,18 +86,24 @@ See the [DC/OS Docker README](https://github.com/dcos/dcos-docker/blob/master/RE
...
+```sh
 make EE_ARTIFACT_URL=<http://...> download-artifacts
 ```
 
 The DC/OS Enterprise artifact is required for some tests.
 
+A license key is required for some tests:
+
+```sh
+cp /path/to/license-key.txt /tmp/license-key.txt
```
Comment (jimenez) [inline CONTRIBUTING.md]: s/license-key.txt/license.txt

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -129,6 +135,40 @@ Linting and some tests are run on Travis CI.
 See `.travis.yml` for details on the limitations.
 To check if a new change works on CI, unfortunately it is necessary to change `.travis.yml` to run the desired tests.
 
+### Rotating license keys
+
+DC/OS Enterprise requires a license key.
+Mesosphere uses license keys internally for testing, and these expire regularly.
+A license key is encrypted and used by the Travis CI tests.
+
+To update the license key, put a file with the contents to use at `license-key.txt` in the root directory of a clone of this repository.
```
Comment (jimenez) [inline CONTRIBUTING.md]: s/license-key.txt/license.txt

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -129,6 +135,40 @@ Linting and some tests are run on Travis CI.
...
+Do not share this file or push it to GitHub.
+Encrypt this file and push the encrypted file to GitHub.
+
+```sh
+travis encrypt-file license-key.txt --add --force
```
Comment (jimenez) [inline CONTRIBUTING.md]: s/license-key.txt/license.txt

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -129,6 +135,40 @@ Linting and some tests are run on Travis CI.
...
+git add license-key.txt.enc .travis.yml
```
Comment (jimenez) [inline CONTRIBUTING.md]: s/license-key.txt/license.txt

File: README.md [hunk 8]
```diff
@@ -73,6 +73,7 @@ def test_oss_example():
 def test_ee_example():
 
     ee_artifact = Path('/tmp/dcos_generate_config.ee.sh')
+    license_key_contents = Path('/tmp/license-key.txt').read_text()
```
Comment (jimenez) [inline README.md]: s/license-key.txt/license.txt

File: tests/conftest.py [hunk 9]
```diff
@@ -38,3 +38,11 @@ def oss_artifact_url() -> str:
...
     """
     return 'https://downloads.dcos.io/dcos/stable/dcos_generate_config.sh'
+
+
+@pytest.fixture(scope='session')
+def license_key_contents() -> str:
+    """
+    Return a license key suitable for use with the latest version of DC/OS.
+    """
+    return Path('/tmp/license-key.txt').read_text()
```
Comment (jimenez) [inline tests/conftest.py]: s/license-key.txt/license.txt

**Comment by adamtheturtle** [comment]
Thank you for the review @jimenez .

However, I don't agree - the file is a key, and not key contents. The key contents are the return value of the fixture, the `read_text()` result on the file.

**Comment by adamtheturtle** [comment]
The build is showing `The license key is expired or invalid. Please contact support.`.

**Pull request merged**